fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1091,25 +1091,210 @@ inline bool tenantMapChanging(MutationRef const& mutation) {
 	return false;
 }
 
+// return the first tenantId whose idToPrefix(id) >= prefix[0..8] in lexicographic order. If no such id, return
+// INVALID_TENANT;
+inline int64_t lowerBoundTenantId(const StringRef& prefix, const std::map<int64_t, TenantName>& tenantMap) {
```

Comment:
There is a function `TenantIdCodec::lowerBound` that could possibly help with this. It's slightly different in that it returns the first tenant ID not smaller than the encoding for the current key, but if you passed in the first up to 8 bytes of the string, it would give you the corresponding ID.